File: src/sort/bubbleSort.ts
export default function bubbleSort(arr: Array<number>): Array<number> {
  let swapped = false;
  let count = 0;

  while (swapped || count === 0) {
    swapped = false;
    count++;

    for (let i = 0; i < arr.length - count; i++) {
      let j = arr[i];
      let k = arr[i + 1];

      if (typeof k !== "undefined" && j > k) {
        arr[i] = k;
        arr[i + 1] = j;
        swapped = true;
      }
    }
  }

  return arr;
}

File: src/sort/bubbleSort.go
package sort

func BubbleSort(arr []int) []int {
	swapped := true

	for swapped == true {
		swapped = false
		
		for i := 0; i < len(arr); i++ {
			if i < len(arr) - 1 && arr[i] > arr[i+1] {
				temp := arr[i]
				arr[i] = arr[i+1]
				arr[i+1] = temp
				swapped = true
			}
		}
	}

	return arr;
}